bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,401 @@
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            try {
+                ContentResponse response = httpClient.GET(generatePingUrl());
+                if (response.getStatus() == 200) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            response.getContentAsString());
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 5, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private static String getMeasureId(Measure measure) {
+        String id = measure.getLocationId();
+        if (id.isEmpty()) {
+            // Local devices don't have location ID.
+            id = measure.getSerialNo();
+        }
+
+        return id;
+    }
+
+    protected void pollingCode() {
+        List<Measure> measures = getMeasures();
+        Map<String, Measure> measureMap = measures.stream().collect(Collectors.toMap((m) -> getMeasureId(m), (m) -> m));
+
+        for (Thing t : getThing().getThings()) {
+            AirGradientLocationHandler handler = (AirGradientLocationHandler) t.getHandler();
+            if (handler != null) {
+                String locationId = handler.getLocationId();
+                @Nullable
+                Measure measure = measureMap.get(locationId);
+                if (measure != null) {
+                    handler.setMeasurment(locationId, measure);
+                } else {
+                    logger.debug("Could not find measures for location {}", locationId);
+                }
+            }
+        }
+    }
+
+    /**
+     * Return location ids we already have things for.
+     * 
+     * @return location ids we already have things for.
+     */
+    public List<String> getRegisteredLocationIds() {
+        List<Thing> things = getThing().getThings();
+        List<String> results = new ArrayList<>(things.size());
+        for (Thing t : things) {
+            AirGradientLocationHandler handler = (AirGradientLocationHandler) t.getHandler();
+            if (handler != null) {
+                results.add(handler.getLocationId());
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Return list of measures from AirGradient API.
+     *
+     * @return list of measures
+     */
+    public List<Measure> getMeasures() {
+        try {
+            ContentResponse response = httpClient.GET(generateMeasuresUrl());
```

Comment:
Can you add a default timeout to the request?

Reply:
Added to all requests.